File: shinchoku.py
#! /usr/bin/python
# coding: utf-8

import random

def shinchoku(n):
    if n == 0:
        print("進", end='')
        yield "進"
    elif n == 1:
        print("捗", end='')
        yield "捗"
    elif n == 2:
        print("な", end='')
        yield "な"
    elif n == 3:
        print("い", end='')
        yield "い"
    elif n == 4:
        print("で", end='')
        yield "で"
    elif n == 5:
        print("す", end='')
        yield "す"
    else:
        pass

ct = 0
word = []
result = ''
while True:
    l = shinchoku(random.randint(0, 5))
    word.append(next(l))
    if ct > 5:
        word.pop(0)
    result = "".join(word)
    ct += 1

    if result == '進捗ないです':
        print('\nかなしみの', ct, "文字")
        break

File: shinchoku.js
function shinchoku(n){
    if(n == 0)
        return "進";
    else if(n == 1)
        return "捗";
    else if(n == 2)
        return "な";
    else if(n == 3)
        return "い";
    else if(n == 4)
        return "で";
    else if(n == 5)
        return "す";
    else
        return false;
}

function kanashimi(){
    var ct = 0;
    var word = [];
    var res = ['進', '捗', 'な', 'い', 'で', 'す'];
    var tsurai = setInterval(function(){
        var tmp = shinchoku(Math.floor(Math.random() * 6))
        document.write(tmp);
        word.push(tmp);
        ct++;
        if(word.length > 6)
            word.shift();
        if(word.join() == res.join()){
            document.write("<br />");
            document.write("かなしみの" + ct + "文字<br />");
            clearInterval(tsurai);
        }
    }, 30);
};
